src/main/java/am/ik/yavi/core/ConstraintContext.java

```diff
@@ -87,4 +71,47 @@ public String name() {
 			}
 		};
 	}
+
+	@FunctionalInterface
+	interface Attribute {
+		/**
+		 * Returns the attribute value. <code>null</code> is returned if the attribute
+		 * does not exit.
+		 * @return value
+		 */
+		@Nullable
+		Object value();
+
+		/**
+		 * Returns the typed attribute value
+		 * @param clazz the type of the attribute value
+		 * @return value
+		 */
+		@Nullable
+		@SuppressWarnings("unchecked")
+		default <T> T value(Class<T> clazz) {
+			final Object attribute = this.value();
+			if (attribute == null && clazz == Boolean.class) {
+				return (T) Boolean.valueOf(null);
```

Comment:
I had to look twice to understand this line of code. Isn't it better to create a separate `booleanValue()` method instead?

Reply:
Added this code so that a null pointer exception does not occur when evaluating `.value(Boolean.class)`. Now that `.isEqualTo(true)` is available, will remove this code to avoid extra confusion.